### PR TITLE
[TTAHUB-1823] "Topics in AR" graph in RTR (TTA History tab) has incorrect calculations

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
The scope needed to be incorporated into the literal subquery and the reports needed to be deduped across goals.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1823

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: yarn-audit-known-issues [hunk 1]
```diff
@@ -1 +1 @@
...sdk>xml2js","dev":false,"bundled":false,"optional":false},"advisory":{"findings":[{"version":"0.4.19","paths":["aws-sdk>xml2js"]}],"metadata":null,"vulnerable_versions":"<=0.4.23","module_name":"xml2js","severity":"high","github_advisory_id":"GHSA-776f-qx25-q3cc","cves":["CVE-2023-0842"],"access":"public","patched_versions":"<0.0.0","cvss":{"score":7.3,"vectorString":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:L"},"updated":"2023-04-07T21:00:55.000Z","recommendation":"None","cwe":["CWE-1321"],"found_by":null,"deleted":null,"id":1091647,"references":"- https://nvd.nist.gov/vuln/detail/CVE-2023-0842\n- https://fluidattacks.com/advisories/myers/\n- https://github.com/Leonidas-from-XIV/node-xml2js/\n- https://github.com/advisories/GHSA-776f-qx25-q3cc","created":"2023-04-05T21:30:24.000Z","reported_by":null,"title":"xml2js is vulnerable to prototype pollution ","npm_advisory_id":null,"overview":"xml2js version 0.4.23 allows an external attacker to edit or add new properties to an object. This is possible because the application does not properly validate incoming JSON keys, thus allowing the `__proto__` property to be edited.","url":"https://github.com/advisories/GHSA-776f-qx25-q3cc"}}}
```
Comment (thewatermethod) [inline yarn-audit-known-issues:1]: you'll want to merge main in, and probably regenerate this file, as there is a dependabot PR to fix grpc that I just merged in

**Review by thewatermethod — COMMENTED** [review]
Code looks good, could we add a test to prove out the case we are trying to fix?

**Review by AdamAdHocTeam — APPROVED** [review]
Much better! 

Would just add a test as Matt suggested.

File: src/scopes/utils.ts [hunk 2]
```diff
@@ -109,3 +109,33 @@ export function filterAssociation(baseQuery, searchTerms, exclude, callback, com
...
+  model,
+  alias: string,
+  scope: WhereOptions,
+): Promise<string> => {
+  let sql = '';
+  // The db is not connected for this query as the limit is set to zero, it just returns.
+  await model.findAll({
+    where: scope,
+    limit: 0,
+    logging: (x) => { sql = x; },
```
Comment (thewatermethod) [inline src/scopes/utils.ts:131]: Sanity check - all the input ends up escaped right?

**Review by thewatermethod — APPROVED** [review]
Thanks for adding the test

File: src/scopes/utils.ts (hunk 2 above)
Comment (GarrettEHill) [reply src/scopes/utils.ts:131]: I don't understand your question.

Comment (thewatermethod) [reply src/scopes/utils.ts:131]: nvm - I was trying to think if there was something that wouldn't be escaped, but I guess if you are just grabbing the SQL, everything should already have been take care of

**Pull request merged**